packages/stream/create-stream/create-stream.test.js

```diff
@@ -0,0 +1,252 @@
+          init({
+            api_key: 'eyJhbGciOiJ.eyJleHAi.B01hriveOMR',
+          }),
+          config
+        )
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('Error')
+        expect(error.message).to.be.equal(
+          'Failed to create a new stream because the description of the stream is not in string format. A description must be in string format'
+        )
+      }
+    })
+
+    it('should return error response if API Key is not valid', async function () {
+      nock(`${Internal.config.api.base_url}`)
+        .post(`/${Internal.config.api.version}/streams/create`, {
+          name: 'tes',
+          slug: 'tes',
+          description: 'tes',
+        })
+        .reply(403, { code: 403, message: 'API Key is not valid', data: '' })
+
+      const config = { name: 'shopping' }
+      try {
+        await createStream(
+          init({
+            api_key: 'eyJhbGciOiJ.eyJleHAi.B01hriveOMR',
+          }),
+          config
+        )
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('Error')
+        expect(error.message).to.be.equal(
+          'Failed to create a new stream because the API Key is not valid. Please provide a valid and active API Key.'
+        )
+      }
+    })
+
+    it('should return error response if unexpected error from server', async function () {
+      nock(`${Internal.config.api.base_url}`)
+        .post(`/${Internal.config.api.version}/streams/create`, {
+          name: 'tes',
+          slug: 'tes',
+          description: 'tes',
+        })
+        .reply(500, { code: 500, message: 'Server error', data: '' })
+
+      const config = { name: 'shopping' }
+      try {
+        await createStream(
+          init({
+            api_key: 'eyJhbGciOiJ.eyJleHAi.B01hriveOMR',
+          }),
+          config
+        )
+      } catch (error) {
+        expect(error).to.be.an('error')
+        expect(error.name).to.be.equal('Error')
+        expect(error.message).to.be.equal(
+          'Failed to create a new stream because unexpected error from the server'
+        )
+      }
+    })
+  })
+
+  describe('Positive test', function () {
+    const config = { name: 'tes', slug: 'tes', description: 'tes' }
+    beforeEach(function () {
+      nock(`${Internal.config.api.base_url}`)
+        .post(`/${Internal.config.api.version}/streams/create`, config)
+        .reply(200, {
+          status: {
+            code: 200,
+            message: 'Success',
+            type: 'success',
+          },
+          data: {
+            id: 371,
+            name: 'a new stream',
+            slug: '',
+            hls_manifest_path: '',
+            dash_manifest_path: '',
+            description: '',
+            created_by: 6,
+            created_at: '2022-09-05T07:38:09.525329768Z',
+            updated_at: '2022-09-05T07:38:09.525329768Z',
+            quality: '360',
+          },
```

Comment:
@faiq-naufal the data on the test case is just a mock data mas, which I think it's just to check if the data will have those property. 
Some missing properties like `start_time` or `end_time`, sometimes the initial data value is null (from the endpoint), but if I add it into this mock data on `test.js`, it will show error "`use undefined instead of null`". 

Kindly advise, should I just add the missing properties with null value (just ignore the error)?

Reply:
> it will show error errorLens extension "use undefined instead of null".

This is probably from the eslint plugin unicorn rule. I intend to turn this rule off because sometimes we need to set the value to be null and not undefined and this rule complains about that.

Reply:
@faiq-naufal done mas :
- Add eslint plugin unicord null rule at commit [677994a](https://github.com/inlivedev/inlive-js-sdk/pull/11/commits/677994afdc428ea001181ae1ca00da98127d6ff4)
- Add properties on create stream nock mock data at commit [16ad0b6](https://github.com/inlivedev/inlive-js-sdk/pull/11/commits/16ad0b659f2621e724860fa68fd17509208f9328). I add checking too because the return data from created stream some of them are null value.
- Add properties on get stream nock mock data at commit [ec5e467](https://github.com/inlivedev/inlive-js-sdk/pull/11/commits/ec5e4675a3ee1289b25470d100ad884171ee6ed5) -- punten sekalian
- Add properties on get streams nock mock data at commit [f1e68de](https://github.com/inlivedev/inlive-js-sdk/pull/11/commits/f1e68de59c2a8b0a3eb9ecdd49e57eb420b1337c) -- punten sekalian